examples/experimental/dagster-dlift/dagster_dlift/env_client.py

```diff
@@ -0,0 +1,64 @@
+from typing import Any, Mapping, Sequence, Type, Union
+
+from dagster import AssetCheckResult, AssetMaterialization
+from dagster._record import record
+
+from dagster_dlift.client import DbtCloudClient
+from dagster_dlift.translator import (
+    DagsterDbtCloudRuntimeTranslator,
+    DagsterDbtCloudTranslator,
+    DbtCloudProjectEnvironmentData,
+)
+
+
+@record
+class DbtCloudJobRun:
+    """Represents a dbt Cloud job run."""
+
+    job_id: int
+    run_id: int
+    client: DbtCloudClient
+    translator: DagsterDbtCloudTranslator
+    env_data: DbtCloudProjectEnvironmentData
+
+    def wait_for_success(self) -> int:
+        return self.client.poll_for_run_completion(self.run_id)
+
+    def get_run_results(self) -> Mapping[str, Any]:
+        return self.client.get_run_results_json(self.run_id)
+
+    def get_asset_events(
+        self,
+        runtime_translator_cls: Type[
+            DagsterDbtCloudRuntimeTranslator
+        ] = DagsterDbtCloudRuntimeTranslator,
+    ) -> Sequence[Union[AssetMaterialization, AssetCheckResult]]:
+        runtime_translator = runtime_translator_cls()
+        self.wait_for_success()
+        run_results = self.get_run_results()
+        asset_events = []
+        for result in run_results["results"]:
+            data = self.env_data.get_from_unique_id(result["unique_id"])
+            spec = self.translator.get_spec(data)
+            asset_events.append(runtime_translator.get_asset_event(spec, result))
+        return asset_events
+
+
+@record
+class DbtCloudEnvClient:
```

Comment:
nit: maybe `EnvironmentScopedDbtCloudClient`? or `EnvScopedDbtCloudClient`? Right now the distinction between this and the base client is a little unclear

also wondering if this should expose the same methods that `DbtCloudClient` does (& just defer to the nested client in that case), similar to what we have set up with the different scoped host instances in internal - `env_client.whatever()` instead of `env_client.dbt_client.whatever()`

Reply:
I think the key thing is that the user probably shouldn't be using the unscoped client most of the time.

Reply:
but I agree that a rename is probably in order

Reply:
I renamed this one to EnvScopedDbtCloudClient, and the other one to UnscopedDbtCloudClient